picocli-core-module/build.gradle

```diff
@@ -0,0 +1,48 @@
+plugins {
+    id 'java'
+    id 'distribution'
+    id 'maven-publish'
+    id 'com.jfrog.bintray'
+    id 'java-library'
+    id 'org.beryx.jar' version '1.1.3'
+}
+
+group 'info.picocli'
+description 'Picocli JPMS module.'
+version "$projectVersion"
+ext.moduleName = 'info.picocli'
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+
+task copyRootProjectCode(type: Copy) {    
+    from("${rootProject.projectDir}/src/main/java"){
+        exclude '**/groovy'
+    }
+    into "${projectDir}/src/main/java"
+}
+
+compileJava {
+    inputs.property("moduleName", moduleName)   
+    doFirst {
+        options.compilerArgs = [
+            '--module-path', classpath.asPath,
+        ]
```

Comment:
Looks like this fails the build on Java 8... I see this:

```
> Task :picocli-core-module:compileJava FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':picocli-core-module:compileJava'.
> invalid flag: --module-path
```

Maybe we can work around that with something like this:

```
doFirst {
    if (org.gradle.api.JavaVersion.current().isJava9Compatible()) {
        options.compilerArgs = [
            '--module-path', classpath.asPath,
        ]
    }
    classpath = files()
}

Reply:
Java 8 and below won't understand the --module-path argument so that makes sense to me...

Reply:
If this core module is aimed squarely at the JPMS features, why do we need to support java 8 anyway as it the main jar will support java 8?

Reply:
The plugin that’s used allows us to build a Java 9 module with just Java 8. It uses a byte code manipulation library for this. That makes it easier for contributors to set up the environment and build the project so I like that.

Reply:
Cool. I see. Nice work

Reply:
@jwheeler, I'm not in this weekend, feel free to update the gradle file as remkop suggested. 😊